### ISSUE TITLE
Re-destroying throws `module.alb.output.target_group_arns: slice: to index must be <= length of the input list`

### ISSUE THREAD

**Issue by kvz** [terraform-aws-modules/terraform-aws-alb#84]
**Describe the bug**

Re-destroying throws `module.alb.output.target_group_arns: slice: to index must be <= length of the input list`.

**To Reproduce**

1. Add an ALB and set `target_groups_count = "1"`
2. Run a destroy
3. Run a second destroy

See e.g.:

```bash
data.null_data_source.lambda_archive: Refreshing state...
data.null_data_source.lambda_file: Refreshing state...
data.null_data_source.lambda_archive: Refreshing state...
data.null_data_source.lambda_file: Refreshing state...
data.archive_file.notify_slack: Refreshing state...
data.archive_file.notify_slack: Refreshing state...
data.aws_elb_service_account.default: Refreshing state...
data.aws_route53_zone.api2vpc2: Refreshing state...
data.aws_availability_zones.available: Refreshing state...
data.aws_ami.ubuntu: Refreshing state...
data.aws_iam_policy_document.assume_role: Refreshing state...
data.aws_ami.ubuntu: Refreshing state...
data.aws_region.current: Refreshing state...
data.aws_iam_policy_document.lambda_basic: Refreshing state...
data.aws_region.current: Refreshing state...
data.aws_caller_identity.eu-west-2: Refreshing state...
data.aws_iam_policy_document.notify_slack_role_doc: Refreshing state...
data.aws_caller_identity.us-east-2: Refreshing state...
data.aws_region.current: Refreshing state...
data.aws_elb_service_account.default: Refreshing state...
data.aws_route53_zone.api2vpc2: Refreshing state...
data.aws_region.current: Refreshing state...
data.aws_iam_policy_document.notify_slack_role_doc: Refreshing state...
data.aws_ami.ubuntu: Refreshing state...
data.aws_ami.ubuntu: Refreshing state...
data.aws_iam_policy_document.lambda_basic: Refreshing state...
data.aws_availability_zones.available: Refreshing state...
data.aws_iam_policy_document.assume_role: Refreshing state...

Error: Error applying plan:

2 error(s) occurred:

* module.region_us-east-2.module.connectivity.module.alb.output.target_group_arns: slice: to index must be <= length of the input list in:

${slice(concat(aws_lb_target_group.main.*.arn, aws_lb_target_group.main_no_logs.*.arn), 0, var.target_groups_count)}
* module.region_eu-west-2.module.connectivity.module.alb.output.target_group_arns: slice: to index must be <= length of the input list in:

${slice(concat(aws_lb_target_group.main.*.arn, aws_lb_target_group.main_no_logs.*.arn), 0, var.target_groups_count)}
```

Set `target_groups_count = "0"`, run another destory, see:

```bash
Destroy complete! Resources: 0 destroyed.
```

**Expected behavior**

I'd like destroys to be idempotent. So far they have been but the alb module is preventing this now it seems. Although to be fair, it could also very well be that the problem exists between keyboard and chair :) If anything, I hope to learn from submitting this issue.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04.1 LTS
 - Versions: Terraform `v0.11.8`, alb module `3.4.0`, aws provider `1.35.0`

**Additional context**

Thanks for an awesome module! :100:

**Comment by ScriptMyJob** [comment]
Here is a workaround that I did in my terraform.  I am pretty sure that similar logic can be applied in your Terraform IaC as a temp fix while waiting for the module to be updated:

```
variable "global" {
    type = "map"
    default {
        Environment = "staging"
        Project     = "something"
        ServiceName = "something-long-here"
    }
}

locals {
    name = "${lookup(var.global,"Project")}-${lookup(var.global,"Environment")}-${lookup(var.global,"ServiceName")}-tg"
    trim = "${join("",slice(split("","${local.name}"), 0, local.len))}"
    eval = "${length(split("",local.name))}"
    len  = "${local.eval >= 32 ? 32 : local.eval}"
}

output "trim" {
    value  = "${local.trim}"
}
```
Output: 
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

trim = something-staging-something-long
```

You can probably combine this logic into a oneliner, but I think that would make it really hard to read.

**Comment by antonbabenko** [comment]
This issue has been fixed in [v5.0.0](https://github.com/terraform-aws-modules/terraform-aws-alb/releases/v5.0.0) which has been just released.

Check [UPGRADE-5.0.md](https://github.com/terraform-aws-modules/terraform-aws-alb/blob/master/UPGRADE-5.0.md) for upgrade instructions.
